src/main/java/org/sonarlint/intellij/analysis/DefaultClientInputFile.java

```diff
@@ -93,7 +96,15 @@ private URI createURI() {
 
   @Override public String contents() throws IOException {
     if (doc == null) {
-      return new String(vFile.contentsToByteArray(), charset);
+      ByteArrayOutputStream result = new ByteArrayOutputStream();
+      try (InputStream inputStream = inputStream()) {
+        byte[] buffer = new byte[DEFAULT_BUFFER_SIZE];
+        int length;
+        while ((length = inputStream.read(buffer)) != -1) {
+          result.write(buffer, 0, length);
+        }
+      }
+      return result.toString(charset.name());
```

Comment:
We could also use `IOUtils.toString(InputStream, Charset)`. It's from commons-io but provided through IJ dependency

Reply:
:+1: